op-node/rollup/derive/attributes.go

```diff
@@ -41,17 +40,7 @@ func PreparePayloadAttributes(ctx context.Context, cfg *rollup.Config, dl L1Rece
 				fmt.Errorf("cannot create new block with L1 origin %s (parent %s) on top of L1 origin %s",
 					epoch, info.ParentHash(), l2Parent.L1Origin))
 		}
-		for {
-			if err := receiptsFetcher.Fetch(ctx); err == io.EOF {
-				break
-			} else if err != nil {
-				return nil, NewTemporaryError(fmt.Errorf("failed to fetch more receipts: %w", err))
-			}
-		}
-		receipts, err := receiptsFetcher.Result()
-		if err != nil {
-			return nil, NewResetError(fmt.Errorf("fetched bad receipt data: %w", err))
```

Comment:
With the new change we don't distinguish between a temporary RPC error and a corrupted (or malicious) L1 endpoint providing bad receipt data. I think it's fine as it's probably not the right approach to reorg anyway if we're getting bad L1 data, and we'll follow the right chain by hash eventually.

Reply:
Yea, the other thing is that the iterative batch call will nuke it's local state if it errs on the a failed consistency check & will re-fetch the data. That serves as a better retry logic for when one of the receipts is invalid.